### ISSUE TITLE
Why should we scale the learning rate?

### ISSUE THREAD

**Issue by li-zemin** [uber/horovod#384]
Hello, I run the distributed tensorflow with 2 GPU in one machine correctly. And the horovod simpifies my job greatly. However, I found than the same code is running in these two gpus in parallel, and I just feed  these two training process with distinct batch-data. The gradients of training parameters are computed independently and than averaged and then updated. So why should we scale the learning rate? as the training processes update the parameters independently.

**Comment by alsrgv** [comment]
@li-zemin, scaling learning rate helps improve the model convergence rate. The idea is to scale the learning rate linearly with the batch size to preserve the number of epochs needed for the model to converge, and since the number of synchronous steps per epoch is inversely proportionate to the number of GPUs, training is faster. This approach was publicized in [ImageNet in 1 hour](https://arxiv.org/abs/1706.02677).

**Comment by li-zemin** [comment]
@alsrgv Thans for your reply in time. I read this paper and here is my project code:
```python

import tensorflow as tf
import horovod.tensorflow as hvd

# Initialize Horovod
hvd.init()

# Build model...
loss = ...
opt = tf.train.AdagradOptimizer(0.01 * hvd.size())

# Make training operation
train_op = opt.minimize(loss)

# Add Horovod Distributed Optimizer
opt = hvd.DistributedOptimizer(opt)

# Pin GPU to be used to process local rank (one GPU per process)
config = tf.ConfigProto()
config.gpu_options.visible_device_list = str(hvd.local_rank())
config.gpu_options.allow_growth = True

# initialize a session
sess = tf.Session(config=config)
saver = tf.train.Saver()

sess.run(tf.global_variables_initializer())

# restore the pretrained model
saver.restore(sess, "******.ckpt")

bcast = hvd.broadcast_global_variables(0)

# Variables broadcast
sess.run(bcast)

files = splitFromData()

for epoch in range(start, 15000):
    random.shuffle(files)
    for idx in xrange(0, len(files) / batchsize):
        if idx % 2 == hvd.rank():
            file_batch = files[idx * args.batchsize:(idx + 1) * args.batchsize]
            models, batch_images =make_inputs_and_images_multi_classes(file_batch)
            # training 
            result, _ = sess.run([opt], feed_dict={images: batch_images, real_models: models})
```
In my case, I found that the pre-trained model was restored twice, and the two gpus run  optimize operation independently, they get different training data and run in parallel. 
I would like to make it clear, whether my code is right for the horovod? 
How do I make sure that parameters are updated in an average way that the parameters are updated with average gradients? 
 If the each gpu process the optimize operation independently, should I scale the learning rate?

**Comment by alsrgv** [comment]
@li-zemin, your code is correct.  Processes indeed read the data independently, but they don't do optimization independently -- `opt = hvd.DistributedOptimizer(opt)` command wraps regular TensorFlow optimizer with the Distributed Optimizer which will average gradients among workers before they are applied.  This allows you to scale the learning rate and achieve faster convergence.
